### PR TITLE
Updated school profile with new design

### PR DESCRIPTION
**Pull request by ethax-ross**
### Trello card

[Trello-504](https://trello.com/c/KQx0G24Q/504-develop-profile-now-prototype-is-finished)

### Context

We have a new design for the school profile page; the content is largely identical but restructured.

### Changes proposed in this pull request

- Update helpers with convenience methods

The new design requires a list of subjects and the ability to put DBS check information in the same paragraph as the `Yes/No` so we need to be able to set the `wrapper_tag`.

- Support month/year format

We use this in the placement date headings, e.g. `March 2022`.

- Add has_fees? method

We need to be able to determine if a booking has any fees attached as the new profile design will show/hide the whole section depending on if there are any fees.

- Add styles for content list

The new school profile design has a contents list at the top to easily index into the page. We are taking the styles from GOV.UK publishing components instead of pulling in the whole gem.

- Update with combined placement dates method

The new school profile design shows all placement dates in the same list (primary + secondary). We also show a total date count in the sidebar; this adds the necessary methods to the presenter.

Delegate `has_fees?` for convenience.

- Minor heading update to match new design

Tweaks the heading of the show page to match the new design.

- Rework school profile to match new design

Reworks the school profile section to match the new design; the page content is virtually identical with just ordering/structure updates.

- Update specs to account for new profile design

Update the specs/features so that they run against the new profile design; I've tried to keep the tests the same with mainly selectors changing so that we can be sure we haven't lost functionality with the design change.

- Break phase2 view into partials

To make it easier to scan through and change I'm breaking out the sections into individual partials.

- Anchor to appropriate placement on start request

When a candidate hits the 'start request' link for a school that has fixed placement dates they are taken to another page that lists all the dates and subjects. We want to jump the user down to the appropriate date for the placement they selected.

### Guidance to review

I've tried to do this incrementally but its a bit of a big bang PR as there's no real middle-ground with it.

The new design can be found [here](https://get-school-experience-prototype.london.cloudapps.digital/profile/profile).

I haven't changed the `phase1` design, which appears to be used for schools that are not yet onboarded; I'm not sure if we intend on updating this as well?

| Fixed Dates - Desktop      | Fixed Dates - Mobile | Availability Text - Desktop | Availability Text - Mobile |
| ----------- | ----------- | ----------- | ----------- |
| screencapture-localhost-3000-candidates-schools-142541-2022-02-09-13_51_22      |  screencapture-localhost-3000-candidates-schools-142541-2022-02-09-13_59_29       | screencapture-localhost-3000-candidates-schools-142541-2022-02-09-13_57_36      |  screencapture-localhost-3000-candidates-schools-142541-2022-02-09-13_58_22       |

### CODE REVIEW

**Comment by MylesJarvis** [comment]
Looks good - but let's wait until I'm back to merge this. I also want to speak to Adam H about testing if needed.

**Review by leoapost — APPROVED** [review]

**Review by leoapost — COMMENTED** [review]
I just noticed that profiles of non onboarded school are still using the old design. Is that intended?

review-school-experience-2272 london cloudapps digital_candidates_schools_140234

**Comment by MylesJarvis** [comment]
@leoapost I think we should tackle this in a separate PR once this is merged.

**Pull request merged**
